### PR TITLE
ccip cap config contract updates

### PR DESCRIPTION
**Pull request by makramkd**
## Motivation

The most recent merge introduced breaking changes to the capability registry contract. This PR fixes them.

## Solution

* capability registry -> capabilities registry rename
* implement IERC165 for the capabilities registry
* update ICapabilitiesRegistry interface

### CODE REVIEW

File: contracts/gas-snapshots/ccip.gas-snapshot [hunk 1]
```diff
@@ -34,56 +34,56 @@ BurnWithFromMintTokenPool_lockOrBurn:test_ChainNotAllowed_Revert() (gas: 28789)
...
-CCIPCapabilityConfiguration_beforeCapabilityConfigSet:test_beforeCapabilityConfigSet_ExecConfigOnly_Success() (gas: 1055313)
+CCIPCapabilityConfiguration__updatePluginConfig:test__updatePluginConfig_InvalidConfigStateTransition_Reverts() (gas: 23105)
+CCIPCapabilityConfiguration__updatePluginConfig:test__updatePluginConfig_RunningToStaging_Success() (gas: 2002384)
+CCIPCapabilityConfiguration__updatePluginConfig:test__updatePluginConfig_StagingToRunning_Success() (gas: 2608052)
+CCIPCapabilityConfiguration__updatePluginConfig:test_getCapabilityConfiguration_Success() (gas: 9583)
+CCIPCapabilityConfiguration_beforeCapabilityConfigSet:test_beforeCapabilityConfigSet_CommitAndExecConfig_Success() (gas: 1844013)
+CCIPCapabilityConfiguration_beforeCapabilityConfigSet:test_beforeCapabilityConfigSet_CommitConfigOnly_Success() (gas: 1062710)
+CCIPCapabilityConfiguration_beforeCapabilityConfigSet:test_beforeCapabilityConfigSet_ExecConfigOnly_Success() (gas: 1062741)
 CCIPCapabilityConfiguration_beforeCapabilityConfigSet:test_beforeCapabilityConfigSet_OnlyCapabilityRegistryCanCall_Reverts() (gas: 9576)
```
Comment (0xsuryansh) [inline contracts/gas-snapshots/ccip.gas-snapshot]: nit: For the sake of consistency, I think we should rename the `CapabilityRegistry`, occurrences in test files as well?

https://github.com/smartcontractkit/ccip/blob/update-capconfig-after-cr-updates/contracts/src/v0.8/ccip/test/helpers/CCIPCapabilityConfigurationHelper.sol#L7 


The constructor of this helper `constructor(address capabilityRegistry)`  also has the old parameter name whereas it should be `capabilitiesRegistries` 

Let me know your thoughts

File: contracts/src/v0.8/ccip/capability/CCIPCapabilityConfiguration.sol [hunk 2]
```diff
@@ -3,18 +3,19 @@ pragma solidity 0.8.24;
...
 import {EnumerableSet} from "../../vendor/openzeppelin-solidity/v4.8.3/contracts/utils/structs/EnumerableSet.sol";
 
 /// @notice CCIPCapabilityConfiguration stores the configuration for the CCIP capability.
-/// We have two classes of configuration: chain configuration and DON (in the CapabilityRegistry sense) configuration.
+/// We have two classes of configuration: chain configuration and DON (in the CapabilitiesRegistry sense) configuration.
 /// Each chain will have a single configuration which includes information like the router address.
 /// Each CR DON will have up to four configurations: for each of (commit, exec), one blue and one green configuration.
 /// This is done in order to achieve "blue-green" deployments.
-contract CCIPCapabilityConfiguration is ITypeAndVersion, ICapabilityConfiguration, OwnerIsCreator {
+contract CCIPCapabilityConfiguration is ITypeAndVersion, ICapabilityConfiguration, OwnerIsCreator, IERC165 {
```
Comment (RensR) [inline contracts/src/v0.8/ccip/capability/CCIPCapabilityConfiguration.sol]: should this also be named `capabilities` now?

File: contracts/src/v0.8/ccip/capability/CCIPCapabilityConfiguration.sol [hunk 3]
```diff
@@ -131,9 +132,16 @@ contract CCIPCapabilityConfiguration is ITypeAndVersion, ICapabilityConfiguratio
...
+  /// @param capabilitiesRegistry the canonical capabilities registry address.
+  constructor(address capabilitiesRegistry) {
+    i_capabilitiesRegistry = capabilitiesRegistry;
+  }
+
+  /// @inheritdoc IERC165
+  function supportsInterface(bytes4 interfaceId) external pure override returns (bool) {
+    return interfaceId
+      == ICapabilityConfiguration.getCapabilityConfiguration.selector
+        ^ ICapabilityConfiguration.beforeCapabilityConfigSet.selector;
```
Comment (RensR) [inline contracts/src/v0.8/ccip/capability/CCIPCapabilityConfiguration.sol]: What's this `^` syntax?

File: contracts/src/v0.8/ccip/capability/interfaces/ICapabilitiesRegistry.sol [hunk 4]
```diff
@@ -1,23 +1,31 @@
 // SPDX-License-Identifier: MIT
 pragma solidity ^0.8.24;
 
-interface ICapabilityRegistry {
+interface ICapabilitiesRegistry {
```
Comment (RensR) [inline contracts/src/v0.8/ccip/capability/interfaces/ICapabilitiesRegistry.sol:4]: This feels like it should exist outside of ccip as it's not a ccip specific interface right?

File: contracts/src/v0.8/ccip/capability/CCIPCapabilityConfiguration.sol (hunk 2 above)
Comment (makramkd) [reply contracts/src/v0.8/ccip/capability/CCIPCapabilityConfiguration.sol]: That's reasonable yes.

File: contracts/src/v0.8/ccip/capability/CCIPCapabilityConfiguration.sol (hunk 3 above)
Comment (makramkd) [reply contracts/src/v0.8/ccip/capability/CCIPCapabilityConfiguration.sol]: I believe a byte-by-byte xor - this is the interface check used by the registry https://github.com/smartcontractkit/ccip/blob/9075d9b2aac6036b8a70dbbe6ef8a135574851a2/contracts/src/v0.8/keystone/CapabilitiesRegistry.sol#L1013-L1015

File: contracts/src/v0.8/ccip/capability/interfaces/ICapabilitiesRegistry.sol (hunk 4 above)
Comment (makramkd) [reply contracts/src/v0.8/ccip/capability/interfaces/ICapabilitiesRegistry.sol:4]: Its not ccip specific but it is the narrow interface used by the config contract. Alternatively we can just import the full registry (they don't have an interface for it) but I'm not sure if that's preferable for compile times / contract size etc.

File: contracts/src/v0.8/ccip/capability/CCIPCapabilityConfiguration.sol (hunk 2 above)
Comment (makramkd) [reply contracts/src/v0.8/ccip/capability/CCIPCapabilityConfiguration.sol]: Renamed to CCIPConfig

File: contracts/src/v0.8/ccip/capability/interfaces/ICapabilitiesRegistry.sol (hunk 4 above)
Comment (elatoskinas) [reply contracts/src/v0.8/ccip/capability/interfaces/ICapabilitiesRegistry.sol:4]: We would be able to import this after we merge in chainlink core right? And then we can remove this file

File: contracts/src/v0.8/ccip/capability/CCIPCapabilityConfiguration.sol (hunk 3 above)
Comment (elatoskinas) [reply contracts/src/v0.8/ccip/capability/CCIPCapabilityConfiguration.sol]: Had this question too, seems common: https://ethereum.stackexchange.com/a/40989

**Review by elatoskinas — APPROVED** [review]

File: contracts/src/v0.8/ccip/capability/interfaces/ICapabilitiesRegistry.sol (hunk 4 above)
Comment (elatoskinas) [reply contracts/src/v0.8/ccip/capability/interfaces/ICapabilitiesRegistry.sol:4]: nvm this is the Capability Registry, not the Config interface - I recon we could add this interface under keystone, and have the keystone capability registry implement it

cc @cds95

Comment (makramkd) [reply contracts/src/v0.8/ccip/capability/interfaces/ICapabilitiesRegistry.sol:4]: Created CCIP-2647 to address this

**Review by dimkouv — APPROVED** [review]

**Review by 0xsuryansh — APPROVED** [review]

File: contracts/gas-snapshots/ccip.gas-snapshot (hunk 1 above)
Comment (makramkd) [reply contracts/gas-snapshots/ccip.gas-snapshot]: Done.

**Review by RensR — APPROVED** [review]

**Pull request merged**